From ea34d8dc8d99a20988900070a4d4867fd413c61c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <P@draigBrady.com>
Date: Mon, 5 Mar 2012 14:48:21 +0000
Subject: [PATCH] allow nova-manage to auto support the new cfg format

This is a temprorary work around to support the
new config file format transparently.
---
 bin/nova-manage |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/bin/nova-manage b/bin/nova-manage
index 4927614..a8b819b 100755
--- a/bin/nova-manage
+++ b/bin/nova-manage
@@ -97,6 +97,20 @@ from nova.compute import vm_states
 from nova.db import migration
 from nova.volume import volume_types
 
+# PB: Temp hack to support new config file format
+def handle_new_cfg_format():
+    flagfile=utils.default_flagfile()
+    ff=file(flagfile)
+    ini = '[DEFAULT]' in ff.read(64)
+    ff.close()
+    if ini and not any('--config-file' in i for i in sys.argv):
+        for i in range(len(sys.argv)):
+            if '--flagfile' in sys.argv[i]:
+                del sys.argv[i]
+                break
+        sys.argv.insert(1,'--config-file='+flagfile)
+handle_new_cfg_format()
+
 FLAGS = flags.FLAGS
 flags.DECLARE('flat_network_bridge', 'nova.network.manager')
 flags.DECLARE('num_networks', 'nova.network.manager')
